src/commands/publish/upload_form/wasm_module.rs

```diff
@@ -0,0 +1,27 @@
+use super::binding::{Binding, ToBinding};
+use super::file::{File, ToFile};
+
+#[derive(Debug)]
+pub struct WasmModule {
+    pub path: String,
+    pub filename: String,
+    pub binding: String,
+}
+
+impl ToBinding for WasmModule {
+    fn to_binding(&self) -> Binding {
+        let name = self.filename.clone();
+        let part = self.binding.clone();
```

Comment:
these are the worst... every time i think i know which is which i start to doubt. I will double check and add comments, as these names are super generic.